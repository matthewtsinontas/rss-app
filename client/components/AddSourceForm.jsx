import React from 'react';

/**
  Component for the form to subscribe to a new sources
  Displays a form with a single input which accepts a value with an onSubmit
    method and some loading states
*/
const AddSourceForm = ({newSource, loading, updateNewSource, addNewSource}) => (
  <form onSubmit={addNewSource} className="form-inline">
    <div className="form-group">
      <input
        className="form-control"
        disabled={loading}
        type="text"
        value={newSource}
        onChange={updateNewSource}
        placeholder="Add news source"
      />
    </div>
    <button disabled={loading} type="submit" className="btn btn-success">
      {loading ? "Loading ..." : "Add"}
    </button>
  </form>
);

export default AddSourceForm;
