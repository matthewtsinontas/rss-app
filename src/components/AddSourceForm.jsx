import React from 'react';

const AddSourceForm = ({newSource, loading, updateNewSource, addNewSource}) => (
  <form onSubmit={addNewSource}>
    <input
      disabled={loading}
      type="text"
      value={newSource}
      onChange={updateNewSource}
      placeholder="Add news source"
    />
    <button disabled={loading} type="submit">
      {loading ? "Loading ..." : "Add new source"}
    </button>
  </form>
);

export default AddSourceForm;
