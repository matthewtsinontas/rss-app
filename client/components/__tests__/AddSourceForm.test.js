import React from 'react';
import AddSourceForm from '../AddSourceForm.jsx';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

it('renders with the correct props', () => {
  const item = (
    <AddSourceForm
      newSource="newSource"
      loading={false}
      updateNewSource={() => {}}
      addNewSource={() => {}}
    />
  );
  const tree = renderer.create(item).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should call the provided function when the form is submitted', () => {
  const onFormSubmit = sinon.spy();
  const wrapper = shallow(
    <AddSourceForm
      newSource="newSource"
      loading={false}
      updateNewSource={() => {}}
      addNewSource={onFormSubmit}
    />
  );
  wrapper.find('form').simulate('submit');
  expect(onFormSubmit.calledOnce).toEqual(true);
});

it('should disable the form inputs and display the correct text if loading', () => {
  const wrapper = shallow(
    <AddSourceForm
      newSource="newSource"
      loading={true}
      updateNewSource={() => {}}
      addNewSource={() => {}}
    />
  );
  expect(wrapper.find('button').text()).toEqual("Loading ...");
  expect(wrapper.find('input').html().includes('disabled=""')).toBeTruthy();
  expect(wrapper.find('button').html().includes('disabled=""')).toBeTruthy();

});
