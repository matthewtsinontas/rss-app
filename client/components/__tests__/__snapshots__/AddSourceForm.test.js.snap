// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct props 1`] = `
<form
  className="form-inline"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      disabled={false}
      onChange={[Function]}
      placeholder="Add news source"
      type="text"
      value="newSource"
    />
  </div>
  <button
    className="btn btn-success"
    disabled={false}
    type="submit"
  >
    Add
  </button>
</form>
`;
